google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/ManagedVmCloudType.java

```diff
@@ -148,6 +159,36 @@ public ManagedVmDeploymentConfiguration createDefaultConfiguration(
           new CloudSdkAppEngineHelper(new File(server.getConfiguration().getCloudSdkExecutablePath()),
               server.getConfiguration().getCloudProjectName()));
     }
+
+    protected static class PlaceholderDeploymentSource extends ModuleDeploymentSourceImpl {
```

Comment:
if we go the UserSpecified.. naming route you could also rename this class to UserSpecfiedPathDeploymentSource or some shit like that.